draft-kazuho-httpbis-priority.md

```diff
@@ -469,6 +473,17 @@ Element ID is interpreted as a Push ID.
 Empty:
 : A seven-bit field that has no semantic value.
 
+The PRIORITY_UPDATE frame in HTTP/3 MUST NOT be sent before the stream
+has been opened or a PUSH_PROMISE for the corresponding PUSH_ID has been
+received.
+
+In HTTP/3, the PRIORITY_UPDATE frame can arrive before any data is received
+on the corresponding stream or server push due to reordering or proactively
+changing the priority of a server push.  If the stream ID is one which can
+be opened, the frame SHOULD be buffered until the stream is opened and
+applied immediately after the HEADERS are parsed.  This consumes extra state
+on the peer, but existing stream limits bound the size of that state.
```

Comment:
I think this is _almost_ there in achieving this goal. I think we should split out the error cases because of the difference in the handling of request and push IDs. For example,

```
The HTTP/3 PRIORITY_UPDATE frame is sent on the control stream. Because there are no ordering
guarantees between streams, a client that reprioritizes requests before receiving response data might cause 
the server to receive a PRIORITY_UPDATE for an unknown request. If the request stream ID is within 
bidirectional stream limits, the PRIORITY_UPDATE frame SHOULD be buffered until the stream is opened and 
applied immediately after the request message has been processed. Holding PRIORITY_UPDATES in this way
 consumes extra state on the peer, although the size of state is bounded by bidirectional stream limits there is 
no bound on the number of PRIORITY_UPDATES that a client can send. An endpoint that wishes to minimise
 state might decide to store only the most recently received frame.
```

then we need a paragraph that explain the specifics of the push ID space, which has a few differences.